util.go

```diff
@@ -50,3 +54,73 @@ func generateJSONResponse(rw http.ResponseWriter, toMarshal interface{}) {
 	}
 	rw.Write(responseJSON)
 }
+
+var redisMastersTotal = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "todoapp_redis_masters_total",
+		Help: "Total count of available redis masters",
+	},
+	[]string{"instance", "version"},
+)
+
+var redisMastersHealthyTotal = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "todoapp_redis_masters_healthy_total",
+		Help: "Total count of healthy redis masters",
+	},
+	[]string{"instance", "version"},
+)
+
+var redisSlavesTotal = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "todoapp_redis_slaves_total",
+		Help: "Total count of available redis slaves",
+	},
+	[]string{"instance", "version"},
+)
+
+var redisSlavesHealthyTotal = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "todoapp_redis_slaves_healthy_total",
+		Help: "Total count of healthy redis slaves",
+	},
+	[]string{"instance", "version"},
+)
+
+func registerMetrics() {
+	prometheus.MustRegister(redisMastersTotal)
+	prometheus.MustRegister(redisMastersHealthyTotal)
+	prometheus.MustRegister(redisSlavesTotal)
+	prometheus.MustRegister(redisSlavesHealthyTotal)
+}
+
+func getHealthStatus() (map[string]string) {
+	okString := "ok"
+	result := map[string]string{"self": okString}
+	hostname, err := os.Hostname()
+
+	if err != nil { //TODO we'll just ignore any errors :)
+		hostname = "UNKNOWN"
+	}
+
+	result["redis-master"] = checkConnection(masterConnection, masterPassword, okString)
```

Comment:
I think for the first step this is fine but what we actually want is that we lookup the name and check all addresses (https://golang.org/pkg/net/#LookupHost)